src/charm.py

```diff
@@ -142,38 +145,73 @@ def __init__(self, *args):
     def _on_install(self, event: InstallEvent):
         """Handle the install charm event."""
         if not self._container.can_connect():
-            self.unit.status = WaitingStatus("Waiting to be able to connect to vault unit")
             event.defer()
             return
         self._delete_vault_data()
 
+    def _on_collect_status(self, event: CollectStatusEvent):
```

Comment:
I just find this paradigm so much more confusing than keeping everything in configure. How would we even make sure to keep the order of things in sync between configure and collect status? 

How about checking if the event is a collect status event in configure and adding the status only if? How about not using collect status at all? idk

Reply:
There is no need to keep anything in sync between configure and collect status. Status are now **only** managed in collect status. collect status is fired at the end of every event making it so that we don't have to maintain status anywhere else. 

As mentioned in the description, the alternative is to manage status in multiple event handlers which is much worse.